linkerd/http/retry/src/replay.rs

```diff
@@ -1003,26 +1016,49 @@ mod tests {
         T: http_body::Body + Unpin,
     {
         tracing::trace!("waiting for a body chunk...");
-        let chunk = body
-            .data()
+        let chunk = crate::compat::ForwardCompatibleBody::new(body)
+            .frame()
             .await
-            .map(|res| res.map_err(|_| ()).unwrap())
+            .expect("yields a result")
+            .ok()
+            .expect("yields a frame")
+            .into_data()
+            .ok()
             .map(string);
         tracing::info!(?chunk);
         chunk
     }
 
-    async fn body_to_string<T>(mut body: T) -> String
+    async fn body_to_string<B>(body: B) -> (String, Option<HeaderMap>)
```

Comment:
now that we have a single `frame()` method, this function knows the body is done when it gets either a `None` or a `Some(Ok(trailers))`. as a result, we return a `(data, trls)` tuple here.